src/Umbraco.Core/Models/CacheInstruction.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace Umbraco.Cms.Core.Models
+{
+    /// <summary>
+    /// Represents a cache instruction.
+    /// </summary>
+    [Serializable]
+    [DataContract(IsReference = true)]
+    public class CacheInstruction
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CacheInstruction"/> class.
+        /// </summary>
+        public CacheInstruction(int id, DateTime utcStamp, string instructions, string originIdentity, int instructionCount)
+        {
+            Id = id;
+            UtcStamp = utcStamp;
+            Instructions = instructions;
+            OriginIdentity = originIdentity;
+            InstructionCount = instructionCount;
+        }
+
+        /// <summary>
+        /// Cache instruction Id.
+        /// </summary>
+        public int Id { get; private set; }
```

Comment:
Do these need to be private set since they are only set in the ctor? Could just be { get; } ?

Reply:
They could yes, have made that change.